libsolidity/codegen/LValue.cpp

```diff
@@ -216,22 +216,38 @@ void StorageItem::retrieveValue(SourceLocation const&, bool _remove) const
 void StorageItem::storeValue(Type const& _sourceType, SourceLocation const& _location, bool _move) const
 {
 	CompilerUtils utils(m_context);
+	solAssert(m_dataType, "");
```

Comment:
Has the stack element already been converted?  In that case, the clean up should have done there.